firebase-common/src/main/java/com/google/firebase/internal/DataCollectionConfigStorage.java

```diff
@@ -61,19 +60,26 @@ public boolean isEnabled() {
     return dataCollectionDefaultEnabled.get();
   }
 
-  public void setEnabled(boolean enabled) {
+  private synchronized void updateDataCollectionDefaultEnabled(boolean enabled) {
     if (dataCollectionDefaultEnabled.compareAndSet(!enabled, enabled)) {
```

Comment:
Since the method is now synchronized there is no point for the bool to be atomic. Also very important to make isEnabled() synchronized as well.